src/langchain_google_memorystore_redis/chat_message_history.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import redis
+from typing import List, Optional
+
+from langchain_core.chat_history import BaseChatMessageHistory
+from langchain_core.messages import (
+    BaseMessage,
+    message_to_dict,
+    messages_from_dict,
+)
+
+
+class MemorystoreChatMessageHistory(BaseChatMessageHistory):
+    """Chat message history stored in a Cloud Memorystore for Redis database."""
+
+    def __init__(
+        self,
+        client: redis.Redis,
+        session_id: str,
+        ttl: Optional[int] = None,
```

Comment:
Any reason to not spell out "time_to_live"? 

Is it worth using the same terminology redis does -- "expire"?

Reply:
I'm okay to different terms but I generally think TTL is more commonly used in the context of Redis - https://redis.io/commands/ttl/ which is a command to understand how many seconds a key can live in Redis.